utils/utils.c

```diff
@@ -845,3 +849,55 @@ int AlmostEqualRelativeAndAbs_double(double A, double B,
 }
 
 /* #undef __USE_XOPEN2K */
+
+// A parallel cumulative sum
+// Output convention is: cumsum[0] = 0; cumsum[N-1] = sum(a[0:N-1]);
+void parallel_cumsum(const int64_t *a, const int64_t N, int64_t *cumsum){
+    #ifdef _OPENMP
+    int Nthread = omp_get_max_threads();
```

Comment:
Should be `omp_get_num_threads`. Plus, the pedantic comment about the plural `Nthreads`/`nthreads`